### ISSUE TITLE
ZeroTierOne installer knocks to microsoft and require Edge webview

### ISSUE THREAD

**Issue by waokdalf** [zerotier/ZeroTierOne#1591]
Hello devs and community

Thanks for your great idea, your product and open source.
I do not even know how to describe it correctly to get an adequate response. I'm just disappointed.

I was struck by two things:

1) ZeroTierOne installer knocks to microsoft server during installation

2) ZeroTierOne now requires Microsoft Edge WebView component. Without webview client app can't open certain window and accordingly user can't connect to network.
Previously there was no need for Edge in your product. The client had a simple interface with network connectivity.

Please explain why this was done. If you can't explain, in that case let's just let the zerotier community know about it.

I have been following the "Black Mirror" trends in the world for a long time and understand where the legs are growing from. I can't know, but I'm guessing one of two things.
Did Microsoft buy the company or do you have a partnership agreement with them?

![zero1](https://user-images.githubusercontent.com/101395860/158198992-d46462e3-2213-4756-8d80-9a6cb39ec21e.png)
![zero2](https://user-images.githubusercontent.com/101395860/158199311-37dbfc85-1fd4-468e-986e-481540bcbd4a.png)

**Comment by adamierymenko** [comment]
We used Webview because it allows us to develop a cross platform UI easily without bundling Electron, which makes the application smaller and cleaner.

The desktop UI app is currently architected as follows:

There is a small tray application written in Rust that always runs. It's very tiny, smaller than the old UI, and does not run a web view.

The web view is launched as the application window when you open the control panel or any other window and dies when you close it. The backend for the web view is also written in Rust and uses the Rust libraries from the Tauri project: https://tauri.studio

It was at the time the best option for doing a decent cross-platform UI. We are still open to evaluating others in the future. Desktop UI is a complete disaster for reasons that unfortunately beyond our control, and we tried to pick the least-bad option.

Most Windows systems already have edge web view on them, so we didn't bundle it to keep download sizes smaller. Maybe we should if people want to install offline.

We were not bought by anyone.
